pkg/controllers/resources/nodes/syncer.go

```diff
@@ -132,17 +139,64 @@ func (s *nodeSyncer) ModifyController(ctx *synccontext.RegisterContext, builder
 			}
 		}()
 		podCache.WaitForCacheSync(ctx.Context)
-		s.podCache = podCache
+		s.unmamangedPodCache = podCache
+
+		// enqueues nodes based on pod phase changes if the scheduler is enabled
+		// the syncer is configured to update node.status.allocatable by summing the consumption of these pods
+		bld.WatchesRawSource(
+			source.Kind(podCache, &corev1.Pod{}),
+			handler.EnqueueRequestsFromMapFunc(func(_ context.Context, object client.Object) []reconcile.Request {
```

Comment:
This is quite complex to understand, could we just do a `handler.Funcs` here instead of `handler.EnqueueRequestsFromMapFunc` + predicates?

Reply:
Don3